feat: Add Anthropic Files API support

Chesars opened the pull request:
## Title

Add Anthropic Files API support

## Relevant issues

Closes #16232

## Pre-Submission checklist

Please complete all items before asking a LiteLLM maintainer to review your PR

- I have Added testing in the https://github.com/BerriAI/litellm/tree/main/tests/litellm directory, Adding at least 1 test is a hard requirement -
https://docs.litellm.ai/docs/extras/contributing_code
- I have added a screenshot of my new test passing locally
- My PR passes all unit tests on https://docs.litellm.ai/docs/extras/contributing_code
- My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🆕 New Feature

## Changes

## Anthropic Files API

Upload files once to Anthropic and reference them by `file_id` in multiple requests—no need to re-upload content each time.

> **Note:** This is different from OpenAI's Files API (which is for Batches/Fine-tuning). Anthropic's Files API is for referencing files in conversations.

## Features Added

- File upload with multipart/form-data support
- File retrieval with metadata
- File listing
- File deletion
- File content download
- Full async/await and synchronous support
- OpenAI-compatible interface (`litellm.acreate_file()`, `litellm.afile_retrieve()`, etc.)

## Usage

```python
import litellm

# 1. Upload file once
file = await litellm.acreate_file(
    file=open("document.pdf", "rb"),
    purpose="messages",
    custom_llm_provider="anthropic"
)

# 2. Use file_id in messages (no re-upload needed)
response = litellm.completion(
    model="anthropic/claude-sonnet-4-5-20250929",
    messages=[{
        "role": "user",
        "content": [
            {"type": "text", "text": "Summarize this document"},
            {"type": "file", "file": {"file_id": file.id, "format": "application/pdf"}}
        ]
    }]
)

# Other operations
files = await litellm.afile_list(custom_llm_provider="anthropic")
info = await litellm.afile_retrieve(file_id=file.id, custom_llm_provider="anthropic")
await litellm.afile_delete(file_id=file.id, custom_llm_provider="anthropic")
```

## Supported Models by File Type

| File Type | Supported Models | Format Value |
|-----------|-----------------|--------------|
| Images | Claude 3+ | `image/jpeg`, `image/png`, `image/gif`, `image/webp` |
| PDFs | Claude 3.5+ | `application/pdf` |
| Plain text | Claude 3.5+ | `text/plain` |
| Other (code execution) | Claude 3.5 Haiku, Claude 3.7+ | varies |

## Important Notes

- The `file_id` obtained from Anthropic **only works with Anthropic Claude models**
- Token costs are the same whether using `file_id` or inline content
- Benefit:  Not re-uploading files across multiple requests (saves bandwidth, not tokens)
- Max file size: 500 MB | Total storage: 100 GB per org

## Implementation Details

- HTTP-direct approach (no Anthropic SDK dependency required)
- Follows existing BaseFilesConfig pattern
- Response transformation: Anthropic format → OpenAI-compatible format
- Extended HTTP handler with multipart upload support

## Files Changed

- `litellm/llms/anthropic/files/transformation.py` (new)
- `litellm/llms/anthropic/files/handler.py` (new)
- `litellm/files/main.py`
- `litellm/llms/custom_httpx/llm_http_handler.py`
- `litellm/files/main.py` - added `"messages"` to purpose Literal
- `litellm/types/llms/openai.py` - added `"messages"` to CREATE_FILE_REQUESTS_PURPOSE- `docs/my-website/docs/providers/anthropic.md`
- `docs/my-website/docs/files_endpoints.md`
- `tests/test_litellm/llms/anthropic/files/` (new - 13 tests)

## Test

**All tests passing:** ✅

krrishdholakia commented on litellm/files/main.py:
```diff
@@ -406,9 +430,55 @@ def file_retrieve(
+                        async_client = client
+
+                    http_response = await async_client.get(url=url, headers=headers)
+                    return _transform_anthropic_file_response(http_response)
```
can we have this inside `anthropic/files` @Chesars

Chesars committed:
feat(anthropic): add Files API support for SDK

Implement Anthropic Files API (upload, retrieve, list, delete, content)
using the BaseFilesConfig provider pattern. Adds multipart form-data
support to BaseLLMHTTPHandler for file uploads.

Chesars committed:
fix: use calendar.timegm for UTC timestamps and reuse AnthropicError

Address Greptile review feedback:
- Use calendar.timegm() instead of time.mktime() for correct UTC parsing
- Import AnthropicError from common_utils instead of redefining it

Chesars committed:
chore: remove unused AnthropicFilesHandler import and instance

Chesars committed:
fix: clarify purpose field docstring in AnthropicFilesConfig

Chesars committed:
fix: remove redundant import os, use Path().name instead
